### ISSUE TITLE
Flox CLI notifies when new version is available

### ISSUE THREAD

**Issue by ghudgins** [flox/flox#1204]
# Acceptance Criteria

**Given** a new version of flox is available
**When** I run `flox ...` (any command)
**Then** I see a message
```
Flox has a new version available. Go to https://flox.dev to get the latest
```

**Comment by jennymahmoudi** [comment]
I'd expect to list the new version number (if possible) and link to the install docs. Where are we publishing release notes? possibly link to those here or on the install docs page.

```
Flox has a new version available: {new_version}. 
Get the latest at https://flox.dev/docs/install-flox/
```

**Comment by robinbrantley** [comment]
@jennymahmoudi
release notes are published here - https://github.com/flox/flox/tree/main/doc/release-notes

**Comment by jennymahmoudi** [comment]
Added a once-per-day limit to the requirements so that we don't annoy people, and avoid having to implement a setting to turn off the notification.

**Comment by jennymahmoudi** [comment]
created https://github.com/flox/floxdocs/issues/542 for the release notes link

**Comment by tomberek** [comment]
- [ ] add the channel into the installers
- [ ] provide reliable mechanism to obtain most recent version
- [ ] implement notification into CLI

**Comment by stahnma** [comment]
@garbas and @tomberek to break out some prerequisite tasks for this which includes infrastructure on where to ping, what channel needs to be looked at, and more. Those infra tasks may come before this.

**Comment by garbas** [comment]
Few things from my side:

- I think users would appreciate it if we should also the old version (like @ghudgins suggested). Users most often don't know which version of flox they are running. And having old version displayed tell me how far behind the latest version of flox I am.

- And I do agree with @jennymahmoudi that we should only show this once a day, BUT only if the updated version is a patch update (the last number in SemVer - `major.minor.patch`). With everything else (`major` and also `minor`) I would print this warning with every command invocation. The reason for this is that we must move users (even if we annoy them) to the latest version of flox. Maybe we will change this in the future, but with the current plan and changes coming in to flox I'd rather annoy our users than having them use an inferior version of flox.

- Release notes are also published here: e.g., https://github.com/flox/flox/releases/tag/v1.0.2

- Probably a better link to how to upgrade flox is https://flox.dev/docs/install-flox/#upgrade-existing-flox-installation. Perhaps we should have a dedicated page in our docs for this. We could even point to specific architecture (e.g., directly to macOS tab if on macOS).

**Comment by ghudgins** [comment]
>The reason for this is that we must move users (even if we annoy them) to the latest version of flox. Maybe we will change this in the future, but with the current plan and changes coming in to flox I'd rather annoy our users than having them use an inferior version of flox.

I think bothering every 24 hours is more than enough to stay relevant without becoming super chatty and annoying. I would abandon a product that was that in-my-face about it We can always dial it up. 

>Probably a better link to how to upgrade flox is...

I think part of the problem with this link is it doesn't actually have the thing you need to download on it. We could consider an additional line like this but it's a bit long:
```
🚀  Flox has a new version available. {old_version} -> {new_version}
Get the latest at https://flox.dev/docs/install-flox/
Learn about upgrading at https://flox.dev/docs/install-flox/#upgrade-existing-flox-installation
```

**Comment by tomberek** [comment]
We're placing a `LATEST_VERSION` into the top-level of a channel:
```
$ curl https://downloads.flox.dev/by-commit/b9c0f8900ab7bc492f06ddb8b9feec5d977fed5a/LATEST_VERSION
1.0.2-18-gb9c0f89
$ curl https://downloads.flox.dev/by-env/nightly/LATEST_VERSION
1.0.2-17-g2a4537a
$ $ curl https://downloads.flox.dev/by-env/stable/LATEST_VERSION
1.0.2
```

If the installer used is from a release env, it will have a variable set FLOX_SENTRY_ENV which will have a value like: "stable" or "qa" which can be used to check for an update. A default fallback if that variable is not set is to check under "stable" to help keep people at least as up to date as stable releases.

**Comment by mkenigs** [comment]
@ghudgins should this be a blocking network call at the start of any command? I.e.:
```
$ flox list
...delay...
🚀  Flox has a new version available

<list output>
```

How long do you want it to hang before giving up?

**Comment by ghudgins** [comment]
absent a better suggestion: lets use the same 2s that telemetry calls do

**Comment by jennymahmoudi** [comment]
> I think part of the problem with this link is it doesn't actually have the thing you need to download on it. We could consider an additional line like this but it's a bit long:
> 
> ```
> 🚀  Flox has a new version available. {old_version} -> {new_version}
> Get the latest at https://flox.dev/docs/install-flox/
> Learn about upgrading at https://flox.dev/docs/install-flox/#upgrade-existing-flox-installation
> ```

Good point about the upgrades @garbas. Since it's just a link further down the same page (and for macOS, the upgrade instructions links you back up to the top of the page), I would rather not include the extra link. 

Either leave it as is OR....
...to address the issue for upgrade instructions, can we detect the user's platform and give the specific command to get the latest?  for example...

if Debian, show:
```
Ugrade to the latest with:
  sudo apt update
  sudo apt --only-upgrade install flox
```
if RPM, show:
```
Upgrade to the latest with 'sudo yum update flox'
```
if macOS, show:
```
Get the latest at https://flox.dev/docs/install-flox/`
```
for all other (nix, WSL), show:
```
Get the latest at https://flox.dev/docs/install-flox/#upgrade-existing-flox-installation
```

**Comment by mkenigs** [comment]
> absent a better suggestion: lets use the same 2s that telemetry calls do

I think that's too long if we do it before the command. But probably reasonable if we do it after

**Comment by ghudgins** [comment]
after sounds good to me

**Comment by mkenigs** [comment]
> Either leave it as is OR.... ...to address the issue for upgrade instructions, can we detect the user's platform and give the specific command to get the latest? for example...

@jennymahmoudi if you think it's worth it, can you create a 2-pt followup ticket?

**Comment by jennymahmoudi** [comment]
@mkenigs great idea
@ghudgins https://github.com/flox/flox/issues/1304 pls prioritize as you see fit

**Comment by jennymahmoudi** [comment]
Verified that the message displays as expected and doesn't show twice. 👍 

Have not tested that the message reappears after 24 hours, but will open a bug if there's any issue with that.
